### PR TITLE
Add service level tests

### PR DESCRIPTION
**Pull request by rahulPalliyalil**
### What
This pr attempts to add service level test based on the following trello ticket

Trello ticket: https://trello.com/c/UffjCvXw/848-create-component-tests-for-search-data-extractor-service

### How to review

The features for the component level test explains the path of the test. 

### Who can review

Anyone except me

### CODE REVIEW

File: features/steps/steps.go [hunk 1]
```diff
@@ -2,63 +2,92 @@ package steps
...
+}
+
+func (c *Component) iShouldReceivePublishedData() error {
+	outputData := <-c.KafkaProducer.Channels().Output
+
+	var data = &models.SearchDataImport{}
+
+	schema.SearchDataImportEvent.Unmarshal(outputData, data)
+
+	if data.CDID != c.inputData.Description.CDID {
```
Comment (cookel2) [inline features/steps/steps.go:83]: This if statement should be replaced with an assert e.g. 
assert.Equal(&c.ErrorFeature, data.CDID, c.inputData.Description.CDID)

File: features/steps/steps.go [hunk 2]
```diff
@@ -2,63 +2,92 @@ package steps
...
+		return fmt.Errorf("Expected CDID: %v Recevived CDID: %v", c.inputData.Description.CDID, data.CDID)
+	}
+
+	if data.DatasetID != c.inputData.Description.DatasetID {
```
Comment (cookel2) [inline features/steps/steps.go:87]: Similarly this if statement should be replaced with assert too e.g.
assert.Equal(&c.ErrorFeature, data.DatasetID, c.inputData.Description.DatasetID)

**Review by cookel2 — CHANGES_REQUESTED** [review]
I've made a few comments. I also noticed that if I change the inputs in the feature file (e.g. if I change the CollectionID to 444 instead of 123) then the tests should fail but they still pass. So those values might not be being read in.

File: features/publish_data.feature [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Feature: Data extractor should listen to the relevant topic and publish extracted data
+    Scenario: When searching for the extracted data I get the expected result
+        Given I send a kafka event to content published topic
+         | URI                | DataType             |   CollectionID  |
+         | some_uri           | some_datatype        |    123          |
+        When The kafka event is processed
+        Then I should receive the published data
```
[the remark this answers is not in the document]
Comment (rahulPalliyalil) [reply features/publish_data.feature:7]: These values are read in but the published data is what we are interested in , which is being passed in the component stage. These values are just for the kafka consumer

File: features/steps/steps.go (hunk 2 above)
Comment (rahulPalliyalil) [reply features/steps/steps.go:87]: Go always suggests not to use external libraries like testify/assert rather just do code there and use the inbuilt code to check for errors. If we look at the various google libraries,auth libraries etc, we would be able to see it .

File: features/steps/steps.go (hunk 1 above)
Comment (rahulPalliyalil) [reply features/steps/steps.go:83]: Go always suggests not to use external libraries like testify/assert rather just do code there and use the inbuilt code to check for errors. If we look at the various google libraries,auth libraries etc, we would be able to see it .

File: features/steps/component.go [hunk 4]
```diff
@@ -20,8 +22,11 @@ const outputFilePath = "/tmp/dpSearchDataExtractor.txt"
 
 type Component struct {
 	componenttest.ErrorFeature
+	inputData     models.ZebedeeData
```
[the remark this answers is not in the document]
Comment (rahulPalliyalil) [reply features/steps/component.go:24]: I would rather remove the component error feature from here , as there is no usecase for this custom error code.

File: features/steps/steps.go [hunk 5]
```diff
@@ -2,63 +2,92 @@ package steps
...
 
-	assert.Equal(c, "Hello, testURL.com", string(content))
+	for _, e := range observationEvents {
+		if err := c.sendToConsumer(e); err != nil {
+			return err
+		}
+	}
 
-	return c.StepError()
+	return nil
```
[the remark this answers is not in the document]
Comment (rahulPalliyalil) [reply features/steps/steps.go:40]: again, the nil is returned as there is no proper usecase for the custom component test error feature.

File: features/steps/steps.go [hunk 6]
```diff
@@ -2,63 +2,92 @@ package steps
...
-		}
-	}
-
-	time.Sleep(300 * time.Millisecond)
+	signals := registerInterrupt()
 
 	// kill application
 	signals <- os.Interrupt
 
+	return err
```
[the remark this answers is not in the document]
Comment (rahulPalliyalil) [reply features/steps/steps.go:73]: again, the nil is returned as there is no proper usecase for the custom component test error feature.

File: features/steps/steps.go (hunk 1 above)
Comment (nshumoogum) [reply features/steps/steps.go:83]: So we use assertions for both unit and component tests across our Go applications and internal libraries. So I would suggest we continue using assertions for consistency.

File: features/steps/component.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (rahulPalliyalil) [reply features/steps/component.go:25]: Fixed in update 1

File: features/steps/steps.go (hunk 1 above)
Comment (rahulPalliyalil) [reply features/steps/steps.go]: Fixed in update 1.

**Review by cookel2 — APPROVED** [review]
looks good

**Pull request merged**